Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -314,6 +314,37 @@ MediaStreamTrack::TrackCapabilities MediaStreamTrack::getCapabilities() const
     return result;
 }
 
+auto MediaStreamTrack::takePhoto(PhotoSettings&& settings) -> Ref<TakePhotoPromise>
+{
+    // https://w3c.github.io/mediacapture-image/#dom-imagecapture-takephoto
+    // If the readyState of track provided in the constructor is not live, return
+    // a promise rejected with a new DOMException whose name is InvalidStateError,
+    // and abort these steps.
+    if (m_ended || m_readyState != State::Live)
```

Comment:
It is a bit odd to have that handling in MediaStreamTrack::takePhoto but not in MediaStreamTrack::getPhotoCapabilities. If we move them all to MediaStreamTrack, it will be easier to keep `Track has ended` string in sync.

m_ended should be sufficient also, no?
If not, we probably want it in step 339.